src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1317,10 +1318,11 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
             TypeSymbol inputType,
             uint inputValEscape,
             bool hasErrors,
-            DiagnosticBag diagnostics)
+            DiagnosticBag diagnostics,
+            bool underIsPattern)
         {
-            const bool permitDesignations = false; // prevent designators under 'not'
-            var subPattern = BindPattern(node.Pattern, inputType, inputValEscape, permitDesignations, hasErrors, diagnostics);
+            bool permitDesignations = underIsPattern; // prevent designators under 'not' except under an is-pattern
```

Comment:
nit: consider removing local and just naming the arguments below.

Reply:
I did this for clarity.

---
In reply to: [430768444](https://github.com/dotnet/roslyn/pull/44513#discussion_r430768444) [](ancestors = 430768444)